lib/tdf3/src/client/index.ts

```diff
@@ -531,8 +544,12 @@ export class Client {
       throw new ConfigurationError('AuthProvider missing');
     }
     const chunker = await makeChunkable(source);
-    if (!allowList) {
+    if (!allowList && this.allowedKases) {
       allowList = this.allowedKases;
+    } else if (this.platformUrl) {
+      allowList = await fetchKeyAccessServers(this.platformUrl);
```

Comment:
Suggestion: This and the other two awaits above (dpopKeys and makeChunkable) could be performed concurrently, e.g. by taking their promises and explicitly performing a await on the `all` of them

Reply:
This is a great suggestion for optimization. 🎉 

One thing I am not sure how to tie together is that fetchKeyAccessServer might also need the dpop headers. 
If the used authProvided requested the token with dpop the server will expect the dpop for the request.
So unwrap keys will need to await dpopKeys and also fetchKeyAccessServer too. 
Still doable, but it might end up `Promise.All([makeChunkable(), fetchKeyAccessServer])` without the `dpopKeys` because fetchkeyAccessServer might need it.

Is that the case? Do we need to add dpop for fetchKeyAccessServer too?

Reply:
This is a good optimization, but not for a vulnerability fix.

Reply:
Filed a follow-up, DSPX-1011